registry/server/api_topics.go

```diff
@@ -185,25 +185,37 @@ func (s *Server) CreateTopic(ctx context.Context, req *pb.CreateTopicRequest) (*
 	// If we're targeting a specific set of brokers by tag, build
 	// a replica assignment.
 	var assignment kafkaadmin.ReplicaAssignment
-	if req.TargetBrokerTags != nil {
+	if req.TargetBrokerTags != nil || req.TargetBrokerIds != nil {
 		// Create a stub map with the provided request dimensions.
 		opts := kafkazk.Populate(
 			req.Topic.Name,
 			int(req.Topic.Partitions),
 			int(req.Topic.Replication),
 		)
 		pMap := kafkazk.NewPartitionMap(opts)
-
 		// Fetch brokers by tag.
 		reqParams := &pb.BrokerRequest{Tag: req.TargetBrokerTags}
 		resp, err := s.ListBrokers(ctx, reqParams)
 		if err != nil {
 			return empty, err
 		}
 
-		targetBrokerIDs := make([]int, len(resp.Ids))
-		for i := range resp.Ids {
-			targetBrokerIDs[i] = int(resp.Ids[i])
+		selectedBrokerIds := make(map[uint32]bool)
+		if req.TargetBrokerIds != nil {
+			for _, id := range req.TargetBrokerIds {
+				selectedBrokerIds[id] = true
+			}
+		}
+
+		var targetBrokerIDs []int
+		for _, id := range resp.Ids {
```

Comment:
Is it intentional that we're combining brokers by tag and brokers by ID? In the case that we specify `req.TargetBrokerIds` but leave `req.TargetBrokerTags` empty, it looks like we still make a `ListBrokers` call (from memory I feel like that could actually return all brokers, but I could be wrong there). Either way we should probably skip the `ListBrokers` call if we don't need brokers by tag.

Reply:
We still need to validate somehow that the brokers requested actually exist, right?
Although it would probably better to not silently just skip the ones that don't exist, but instead return an `INVALID_ARGUMENT`.